### PR TITLE
Fix mkdocs & rtfd build

### PR DESCRIPTION
**Pull request by janbrasna**
The docs are actually not even building for [almost a year](https://readthedocs.org/projects/www-config/builds/?page=4) now:

Screenshot 2024-06-12 at 23 31 51

It stopped without any apparent reason, so probably just adding the yaml config should fix the builds.

This PR also updates the mkdocs version to help with some dependency woes et al. to be super-uptodate to avoid any surprises. (Also to close #548 too.)

The `mkdocs@1.6.0` + `jinja2@3.1.4` log can be verified [here for `1847787`](https://app.netlify.com/sites/mozmeao-wwwconfig-dev/deploys/666a0cf378d9480008d2b07e)…

The `mkdocs@1.6.1` log is [here for `0ea4e2b`](https://app.netlify.com/sites/mozmeao-wwwconfig-dev/deploys/66ec8150d21a250008c3b4e0)…

Building RTD previews for PRs would need to be enabled in its settings first (not sure if that wouldn't first expect working config anyhoo), but since this can't really break more it shouldn't be necessary.

### CODE REVIEW

**Comment by janbrasna** [comment]
FYI mkdocs now supports dark theme starting from 1.6.0/b5, so I can try flipping the auto on if you feel like it'd be worth it…

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1 @@
-mkdocs==1.2.3
-jinja2==3.0.3  # Downpin else mkdocs breaks
+mkdocs==1.6.0
```
Comment (janbrasna) [inline requirements.txt:1]: mkdocs default template has a lot of a11y issues in 1.6.0, let's hold off until 1.6.1 is out…

Comment (janbrasna) [reply requirements.txt]: Updated to v1.6.1 ✅

**Comment by robhudson** [comment]
Just FYI: We're going to sunset this repo shortly once some work with migrating to django-waffle is completed, so I wouldn't spend a lot of time on improvements here.

More info: The static configs and funnelcakes are no longer used. The only thing used currently is waffle switches which will be replace with database-based switches with django-waffle. This this repo will be archived and no longer used.

**Comment by janbrasna** [comment]
@robhudson Yep, I'm aware, worry not — I'm basically trying (along with the sibling PR for gh-pages) to fix the publishing workflow to perhaps allow publishing the actual deprecation info to the docs sites too, if/when needed — not leaving behind two outdated docs deployments for an archived repo dangling around;)

**Review by robhudson — APPROVED** [review]

**Comment by robhudson** [comment]
Ah okay. I prefer regular commits but a merge commit is fine. Thanks for pointing it out.

**Comment by janbrasna** [comment]
Ah yea that one, it was an older conflict with no impact/diff that GH UI resolves this way (:) but don't hesitate to say a word and I'm happy to update/cleanup/rebase/fixup etc. next time; here it won't matter too much given the repo's useful lifetime…

The good news is now it [successfully builds](https://readthedocs.org/projects/www-config/builds/25805186/) so any deprecation info can be actually published there too!;)

**Pull request merged**